cfpb/cfgov-refresh
# Run Python 2.7 and Python 3.6 containers with Docker

This PR adds a Python 3.6 container running cfgov-refresh. As part of this change, the `attach.sh` and `shell.sh` scripts are removed and replaced with updated Docker usage documentation that describes how to attach to or run a shell in the desired container (since there are now two of them).

This PR includes some other minor cleanup of the usage documentation, including a table of contents so that one can navigate directly to the section of the page one is interested in.

Implementationally, I have moved the Python container's `Dockerfile` and `cfgov/entrypoint.sh` to `docker/python2` and added a corresponding `docker/python3`. This seems like a better way to organize our Docker-related files and scripts. The Python containers join the Elasticsearch and docs container in `docker/`.

## Testing

1. Run `docker-compose up`
2. Verify that you can get to the Python 2 container at http://localhost:8000/ and the Python 3 container at http://localhost:8333/
3. Read the updated usage documentation at http://localhost:8888/usage/#docker

## Screenshots

```
python2_1        | Performing system checks...
python2_1        |
python3_1        | Performing system checks...
python3_1        |
python2_1        | System check identified no issues (0 silenced).
python3_1        | System check identified no issues (0 silenced).
python2_1        | May 09, 2019 - 09:53:08
python2_1        | Django version 1.11.20, using settings 'cfgov.settings.local'
python2_1        | Starting development server at http://0.0.0.0:8000/
python2_1        | Quit the server with CONTROL-C.
python3_1        | May 09, 2019 - 09:53:08
python3_1        | Django version 1.11.20, using settings 'cfgov.settings.local'
python3_1        | Starting development server at http://0.0.0.0:8000/
python3_1        | Quit the server with CONTROL-C.
```

## Notes

The name of the Python 2.7 container is changed to `python2`/`cfgov-refresh_python2_1`.

The `attach.sh` script has been apparently broken for a while, as it is trying to attach to a container name that no longer exists (at some point the Python container name changed from `cfgovrefresh_python_1` to `cfgov-refresh_python_1` (and, with this PR, to `cfgov-refresh_python2_1`/cfgov-refresh_python3_1`).

The Docker usage documentation has also been out-of-date for quite a while. I updated it and reorganized it based on common tasks I know *I* need to perform, but I'd love to have others' opinions.

## Checklist
